### ISSUE TITLE
Accessibility - Dataverse(?) - Spreadsheet Viewer

### ISSUE THREAD

**Issue by tander08** [QualitativeDataRepository/dataverse#103]
## Spreadsheet Viewer for Data (when opened in a new window). (Sample Page): https://qualitativedatarepository.github.io/dataverse-previewers/previewers/SpreadsheetPreview.html?fileid=24770&siteUrl=https://data.qdr.syr.edu&datasetid=24756&datasetversion=1.0 

I'm taking a guess here that the code for the spreadsheet viewer is separate from the rest of Dataverse, so I'm listing all the issues for accessibility on the spreadsheet viewer here. 

### Color contrast:
- [ ] All text in blue on the webpage
- [ ] Text in the top row and first 2 columns from the left in the spreadsheet

### "Document must have title element to aid in navigation"
- [ ] Whole Page

### HTML element must have a lang attribute
- [ ] Whole Page

### Image must have alternate Text
- [ ] QDR logo

### Form elements must have labels
- [ ] Whole Page

### All th elements and elements with role=columnheader/rowheader must have data cells they describe
(I'm not sure if this is a website issue, or the way the data was formatted before uploading issue)
- [ ] First Row of Spreadsheet
- [ ] First Column of Spreadsheet

### Document should have one main landmark
- [ ] Whole Page

### All page content should be contained by landmarks
- [ ] QDR Logo
- [ ] Spreadsheet Viewer Logo
- [ ] Grey Box with File Name etc.
- [ ] Spreadsheet
- [ ] Top Row of Spreadsheet
- [ ] First Column of Spreadsheet
- [ ] Bottom Footer
- [ ] Whole Page

## Text Previewer (Sample: https://qualitativedatarepository.github.io/dataverse-previewers/previewers/TextPreview.html?fileid=24769&siteUrl=https://data.qdr.syr.edu&datasetid=24756&datasetversion=1.0) 

Listing only additional issues not listed above

### Ensure that scrollable region has keyboard access
- [ ] Text in Text Viewer

## PDF Previewer (Sample: https://qualitativedatarepository.github.io/dataverse-previewers/previewers/PDFPreview.html?fileid=24765&siteUrl=https://data.qdr.syr.edu&datasetid=24756&datasetversion=1.0)

Additional issues not listed above: 

### All Page Content should be contained by Landmarks
- [ ] Previous/Next Buttons

### Color Contrast:
- [ ] The small grey paragraph text above each ATI header

### Links must have discernable text 
- [ ] The Annotation for Transparent Inquiry (ATI)  header w/ QDR Logo for each entry

### ul and ol must only directly contain li, script, or template elements
- [ ] Box with all the ATI content 
![image](https://user-images.githubusercontent.com/89411909/140177251-f393a4ca-76dd-4131-9911-00126a928fb2.png)

**Comment by qqmyers** [comment]
FWIW: This issue has some things that are common across the existing previewers (which all share some common HTML), so when I fix those, it should work across the whole set.

**Comment by adam3smith** [comment]
Looks like these aren't fixed yet, correct?

**Comment by qqmyers** [comment]
They were - at least the ones from the previewer framework - just not deployed on stage. I updated them now. You may still see issues with, for example, the spreadsheet viewer that come from the table display library, or the pdf library, etc.

**Comment by qqmyers** [comment]
FWIW, these should be independent of Dataverse, so they can be updated in place on prod at any time.

**Comment by adam3smith** [comment]
I'm seeing this (which is flagged as a hidden form field missing a label by WAVE) at the bottom of the code, below the footer:

```
<div id="CopyPasteDiv" style="position: fixed; top: -10000px; left: -10000px;"><textarea class="copyPaste" style="width: 10000px; height: 10000px; overflow: hidden; opacity: 0;"></textarea></div>
```

Doesn't look like this was previously there, nor is it doing anything?

**Comment by adam3smith** [comment]
WAVE is also complaining about a missing table header above the row number. Not sure about best practices here -- could we hide a header (because it's obvious for people visually what this is)?

**Comment by qqmyers** [comment]
FWIW: I believe these errors are both from the table widget in use for the spreadsheet viewer (https://handsontable.com/). I've flagged these in https://github.com/gdcc/dataverse-previewers/issues/4 - also just noted there that the spreadsheet previewer is way behind, using v0.28.4 from ~2016, so it's possible that upgrading will fix/change the issues.
